### ISSUE TITLE
Mentions should use a unique ID for selection

### ISSUE THREAD

**Issue by rdenman** [PDFTron/webviewer-ui#604]
When @mentioning a user, it appears the value property is used to determine which was selected. However, for many use cases, the value prop will be the user's name (as it is what is displayed in the list of mentionables). Name's can very easily NOT be unique.

`WebViewerInstance.MentionsManager. setUserData()` expects an object containing a value, id, and email. Selection should be based off the id to avoid running into problems when mentionables have the same name.

**Comment by ZhijieZhang** [comment]
Thanks for the feedback regarding the new mentions API!

I might miss something obvious but I wasn't very clear about what you mean by `Selection should be based off the id`. Let's say we have the following user data:

```javascript
instance.mentions.setUserData([
  {
    value: 'Zhijie Zhang',
    id: '1',
  },
  {
    value: 'Zhijie Zhang',
    id: '2',
  }
])
```
The user data contains two people with the same name, and I can see that when you type `@` in the textarea these two identical names show up in the suggestion overlay.

If the selection is based off the `id`, does that mean when a user types `1` the suggestion overlay will do a filter and only show the first person?

Or you would like to see the id being displayed in the suggestion overlay as well(like email) for duplicate names so that users know which one is the correct person to mention?

**Comment by rdenman** [comment]
In my particular use case, I have user data as such: `[{ value: 'Rob Denman', id: '1', email: 'rob.denman@gmail.com' }, { value: 'Rob Denman', id: '2', email: 'rob.denman@yahoo.com' }]` (note the same value/name, but different id/email). In the dropdown of mentionables, I see both the user's name and their email, so I can distinguish between them.

However, selecting either of the 2 selections above will add BOTH to the mentions action - it seems to be selecting the user based off value, which in our case is the name. I'd expect that value would generally be a name for mentions, with email and id generally being unique. If I select an option from the mentionable list, it should base it off the id and only return my selection, not every selection that has an identical value field.

**Comment by ZhijieZhang** [comment]
Thanks for the clarification. I think I understand what the issue is now.

I will try to reproduce on my end and see what I can do from there. This might take some time, so please let me know if you have any concerns or tight timelines. Thanks!

**Comment by ZhijieZhang** [comment]
I've pushed a fix for this issue in both `master` and `6.3`.

To try out the fix, you can download the nightly build tomorrow at https://www.pdftron.com/nightly/#stable/

Click the latest date, then scroll to the bottom and you should see `WebViewer-6.3.1_2020-05-22_stable.zip`.

Let me know if it works ok and if you have found any other issues. Thanks!
